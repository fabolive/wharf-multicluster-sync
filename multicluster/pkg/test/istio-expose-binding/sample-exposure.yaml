apiVersion: networking.istio.io/v1alpha3
kind: DestinationRule
metadata:
  name: dest-rule-ServiceA-default-notls
spec:
  host: ServiceA.default.svc.cluster.local
  subsets:
  - name: notls
    trafficPolicy:
      tls:
        mode: DISABLE
---
apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  name: istio-ingressgateway-ServiceA-mynamespace
spec:
  selector:
    istio: ingressgateway
  servers:
    # Union of all ports exposed by this cluster
  - port:
      number: 80
      name: ServiceA-mynamespace-80
      protocol: TLS
    hosts:
    - ServiceA.mynamespace.svc.cluster.global
    tls:
      mode: PASSTHROUGH
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: ingressgateway-to-ServiceA-mynamespace
spec:
  hosts:
  - ServiceA.mynamespace.svc.cluster.global
  gateways:
  - istio-ingressgateway-ServiceA-mynamespace
  tls:
  - match:
    - port: 80
      sniHosts:
      - ServiceA.mynamespace.svc.cluster.global
    route:
    - destination:
        host: ServiceA.mynamespace.svc.cluster.local
        subset: notls
        port:
          number: 80