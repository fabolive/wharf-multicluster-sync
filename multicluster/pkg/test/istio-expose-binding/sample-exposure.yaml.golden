apiVersion: networking.istio.io/v1alpha3
kind: DestinationRule
metadata:
  creationTimestamp: null
  name: dest-rule-ServiceA-default-notls
  namespace: mynamespace
spec:
  host: ServiceA.default.svc.cluster.local
  subsets:
  - name: notls
    trafficPolicy:
      tls: {}
---
apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  creationTimestamp: null
  name: istio-ingressgateway-ServiceA-mynamespace
  namespace: mynamespace
spec:
  selector:
    istio: ingressgateway
  servers:
  - hosts:
    - FooA.mynamespace.svc.cluster.global
    port:
      name: ServiceA-mynamespace-80
      number: 80
      protocol: TLS
    tls:
      mode: SIMPLE
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  creationTimestamp: null
  name: ingressgateway-to-ServiceA-mynamespace
  namespace: mynamespace
spec:
  gateways:
  - istio-ingressgateway-ServiceA-mynamespace
  hosts:
  - FooA.mynamespace.svc.cluster.global
  tls:
  - match:
    - port: 80
      sniHosts:
      - FooA.mynamespace.svc.cluster.global
    route:
    - destination:
        host: ServiceA
        port:
          number: 80
        subset: notls
